### ISSUE TITLE
Ceph not able to find other OSDs

### ISSUE THREAD

**Issue by nick-lehmann** [rook/rook#3348]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

After following the tutorial using three on-premise nodes, ceph is not able to find the other nodes/OSDs and PVCs get stuck.

**Expected behavior:**

The ceph cluster should be healthy and PVC should get created.

**How to reproduce it (minimal and precise):**

- Follow the [kubeadm tutorial](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/). Use the flannel network plugin and allow the master to run pods.
- Follow the [offiical rook quickstart for ceph](https://rook.io/docs/rook/v1.0/ceph-quickstart.html).  I have used the `cluster.yaml` instead of `cluster-test.yml` since I have more than one node.
- Check that everything is working. Using the toolbox, I checked that the status of my cluster was ok, which it was.
- Follow the [instructions for setting up block storage](https://rook.io/docs/rook/v1.0/ceph-block.html).
- Create a PVC, e.g. by `kubectl create -f mysql.yaml` (the included example).
- Check the cluster heatlh again. Doing this yields the output at the end of the issue.

With the `cluster-test.yaml`, which only includes one mon (which ended up on my master node), everything worked as expected. But using three mons, PVCs get stuck and the `ceph` cli also does not show any OSDs online.

One issue might be that the `flannel` network plugin uses a custom IP range for the pod network (`10.244.0.0/16`). Are there any configuration options tell ceph where it can find the other OSDs?

**Environment**: 
* OS: Debian strech on Nodes, Manjaro on client
* Kernel (e.g. `uname -a`): Linux 4.9.0-9-amd64
* Cloud provider or hardware configuration: [Contabo](https://contabo.de/); three VPS; one large as master; two small as workers
* Rook version (use `rook version` inside of a Rook Pod): `v1.0.0-141.g1bf3cfa`
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"archive", BuildDate:"2019-06-08T16:22:01Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

```
> ceph status
  cluster:
    id:     d4271e9e-7715-4ce7-8467-cc5a664d807f
    health: HEALTH_WARN
            Reduced data availability: 100 pgs inactive
 
  services:
    mon: 3 daemons, quorum a,b,c (age 18s)
    mgr: a(active, since 60m)
    osd: 0 osds: 0 up, 0 in
 
  data:
    pools:   1 pools, 100 pgs
    objects: 0 objects, 0 B
    usage:   0 B used, 0 B / 0 B avail
    pgs:     100.000% pgs unknown
             100 unknown
```

```
> kubectl -n rook-ceph get pods
NAME                                  READY   STATUS      RESTARTS   AGE
rook-ceph-agent-dsl4t                 1/1     Running     0          3h56m
rook-ceph-agent-gbfvd                 1/1     Running     0          3h56m
rook-ceph-agent-scp84                 1/1     Running     0          3h56m
rook-ceph-mgr-a-7dd657ffc6-zcqgs      1/1     Running     1          3h56m
rook-ceph-mon-b-765668bdf-s8ndr       1/1     Running     0          175m
rook-ceph-mon-c-746dbd94b4-zmcmp      1/1     Running     0          175m
rook-ceph-operator-56b7cdb77b-nknhl   1/1     Running     0          3h59m
rook-ceph-osd-prepare-server1-jw4pz   0/2     Completed   0          173m
rook-ceph-osd-prepare-server2-sslww   0/2     Completed   0          173m
rook-ceph-osd-prepare-server3-m7dg7   0/2     Completed   1          173m
rook-ceph-tools-7cf4cc7568-n58nc      1/1     Running     0          3h52m
rook-discover-h5wj5                   1/1     Running     0          3h59m
rook-discover-s4482                   1/1     Running     0          3h59m
rook-discover-skpnv                   1/1     Running     0          3h59m
```

**Comment by noahdesu** [comment]
@nick-lehmann could you clarify something for me? you  mention that you follow the setup guide and everything is OK:

> Check that everything is working. Using the toolbox, I checked that the status of my cluster was ok, which it was.

Then you mention that you setup block storage, and reference cluster.yaml (which would have been part a prior step taken in setting up the cluster that you observed to be healthy):

> Follow the instructions for setting up block storage. I have used the cluster.yaml instead of cluster-test.yml.

**Comment by nick-lehmann** [comment]
Thanks, that was worded quite badly, I apologize. I followed the quickstart guide which included creating the cluster (with `cluster.yaml`), then checked its health which was ok, and after that continued with block storage. After creating the first PVC, the health was reported as bad (see output of `ceph status` at the end of the description).

**Comment by BlaineEXE** [comment]
If your nodes don't have any disks on them, Ceph won't have anything to use as storage.

**Comment by noahdesu** [comment]
@BlaineEXE you're referring to `osd: 0 osds: 0 up, 0 in`? Ah, I suppose that as long as no pools exist, then Ceph will report itself healthy if no osd have been created. Yeh, need to consume some disks.

**Comment by nick-lehmann** [comment]
@BlaineEXE Ah ok, I see. I checked using the toolbox and you're right, there are no devices attached and @noahdesu's explanation seems reasonable.

```
> ceph device ls
DEVICE HOST:DEV DAEMONS LIFE EXPECTANCY
```

The cluster was indeed created with `useAllDevices` set to false. I did not think I would need this settings, because I assumed ceph would use the first block device it could find, and since my Kubernetes cluster is running on VPS there is only one device per node.

What would be the right way to add devices? Should I just set `useAllDevices` to true? Is it even possible to use the device where the OS is installed and which should not get formatted? Or do I have to create loopback devices on each of my nodes and point ceph to those? That latter was how I setup my glusterfs cluster in the past.
